kolibri/core/device/utils.py

```diff
@@ -141,3 +111,219 @@ def set_app_key_on_response(response):
     from .models import DeviceAppKey
 
     response.set_cookie(APP_KEY_COOKIE_NAME, DeviceAppKey.get_app_key())
+
+
+def _check_setting(name, available, msg):
+    if name not in available:
+        raise ValueError(msg.format(name))
+
+
+def check_facility_setting(name):
+    AVAILABLE_SETTINGS = [
+        "learner_can_edit_username",
+        "learner_can_edit_name",
+        "learner_can_edit_password",
+        "learner_can_sign_up",
+        "learner_can_delete_account",
+        "learner_can_login_with_no_password",
+        "show_download_button_in_learn",
+    ]
+    _check_setting(
+        name,
+        AVAILABLE_SETTINGS,
+        "'{}' is not a facility setting that can be changed by this command",
+    )
+
+
+def check_device_setting(name):
+    AVAILABLE_SETTINGS = [
+        "language_id",
+        "landing_page",
+        "allow_guest_access",
+        "allow_peer_unlisted_channel_import",
+        "allow_learner_unassigned_resource_access",
+        "name",
+        "allow_other_browsers_to_connect",
+    ]
+    _check_setting(
+        name,
+        AVAILABLE_SETTINGS,
+        "'{}' is not a device setting that can be changed by this command",
+    )
+
+
+def validate_facility_settings(new_settings):
+    # Override any settings passed in
+    for key in new_settings:
+        check_facility_setting(key)
+    return new_settings
```

Comment:
Mainly a conversational note below - not a change request.

One pattern I've started to use occasionally is a list of objects rather than a top-level object. E.g. rather than

```json
{
  "some_setting": "some_value",
  "another_setting": "another_value",
}
```

use 

```json
[
  { "name": "some_setting", "default": "some_value" },
  { "name": "another_setting", "default": "another_value" },
]
```

Benefits are that it's easier to process these files with `filter` and `map`, and it's natural to extend with new metadata.

Reply:
It feels unnecessarily verbose to me - and it's not so hard to do `pick` type behaviours with objects, and can always map on `.items()` if need both the keys and the values.